### ISSUE TITLE
cannot manipulate the database

### ISSUE THREAD

**Issue by ys588281** [3logic/apollo-cassandra#28]
I am new in apollo-cassandra. I connect the cassandra server with an external ip. I change the setting of rpc_address to my public ip in cassandra.yaml. I can successfully connect to cassandra server with external ip. But I cannot save or query. I wonder what happen here. The following is my code.

  apollo.connect(function(err){
                            if(err) {
                                console.log(err);
                                logger.error(err);
                                console.log("cassandra connection failed");
                            }
                            else{
                            //my console will print "cassandra connection success". I think i can connect to it 
                                console.log("cassandra connection success");
                                 //add table schema
                                var ChatLog = apollo.add_model('chatlog',chatLogSchema);
                                // create a new object
                                var message = new ChatLog({
                                    uid: data.sender,
                                    time:data.time,
                                    msgid:data.msgid,
                                    sender:data.sender,
                                    receiver:data.receiver,
                                    group:data.grp,
                                    msgtype:data.msgtype,
                                    content:data.msg.message,
                                    bond_detail:null
                                });
                             //save the object to cassabdra
                            message.save(function(err){
                            if(err)
                                console.log("cassandra save error");
                            else
                                console.log("cassandra save success");
                            });

But no matter what I change my console still print "cassandra save error".

Is there anyone knows that what else did I miss?

**Comment by ramiel** [comment]
Can you report exactly the error? Maybe changing to
```javascript
if(err)
    console.log("cassandra save error", err);
```

**Comment by ys588281** [comment]
Thanks. I printed the error ad solve the problem. But I met another two problems.
1. My table has composite primary key. And in cassandra server I already define table.
CREATE TABLE chatlog(
   uid text,
   msg_id text,
   sender text,
   receiver text,
   PRIMARY KEY (uid,msg_id)
   );
In apollo my code is like this
var  chatLogSchema = {
    "fields":{
        "uid":  { "type": "text"},
        "msg_id":  { "type": "text"},
        "sender":  { "type": "text"},
        "receiver":  { "type": "text"}
    },
    "key":[["uid", "msg_id"]]
    };
But when I execute it. It will output error that the table schema is not matching. Is there anything wrong?

2. I need to store a data that its type in javascript is number. In cassandra I used int  but it did not match. What is the data type should I define in cassandra since I don't have too much choice in javascript. Anyone could help me solve the question?
